### PR TITLE
Report signals through DataStore module like we do in newrelic-riak; …

### PR DESCRIPTION
**Pull request by ydderd**
…some minor cleanup; version bump
Resulting changes in NR: https://rpm.newrelic.com/accounts/10196/applications/104129274/datastores?tw%5Bend%5D=1516315346&tw%5Bstart%5D=1516315002#/overview/All?value=total_call_time_per_minute

Ping - @geoffreyclark 
\cc .- @obrie

### CODE REVIEW

File: lib/newrelic-couchbase/instrumentation/couchbase.rb [hunk 1]
```diff
@@ -1,18 +1,15 @@
 ::Couchbase.module_eval do
   class << self
-    include NewRelic::Agent::MethodTracer
-
     [
+      :new,
```
Comment (obrie) [inline lib/newrelic-couchbase/instrumentation/couchbase.rb:4]: What was the reason for adding `new` here?

File: lib/newrelic-couchbase/instrumentation/bucket.rb [hunk 2]
```diff
@@ -1,34 +1,29 @@
...
+    :run,
+    :design_docs,
+    :save_design_doc,
+    :delete_design_doc,
+    :observe_and_wait,
+    :create_timer,
+    :create_periodic_timer
   ].each do |instruction|
-    add_method_tracer instruction, "Couchbase/Bucket/#{instruction.to_s}"
+    NewRelic::Agent::Datastores.trace self, instruction, "Couchbase"
```
Comment (obrie) [inline lib/newrelic-couchbase/instrumentation/bucket.rb:58]: In the newrelic-riak gem, we're actually tracking the name of the bucket as well.  Do we get that for free or do we need to use the `NewRelic::Agent::Datastores.wrap` api?

Comment (obrie) [reply lib/newrelic-couchbase/instrumentation/bucket.rb:58]: Actually looking at your newrelic link, it looks like we don't get that for free.  In Riak, you can see we say "Riak D find" where "D" is the bucket name.

File: lib/newrelic-couchbase/instrumentation/bucket.rb [hunk 3]
```diff
@@ -1,34 +1,60 @@
+module NewRelic
+  module Agent
+    module Instrumentation
+      module Couchbase
+        def get_set_callback(statement)
+          Proc.new do |result, scoped_metric, elapsed|
+            NewRelic::Agent::Datastores.notice_statement(statement, elapsed)
+          end
```
Comment (obrie) [inline lib/newrelic-couchbase/instrumentation/bucket.rb:8]: This should be a constant rather than generated on every call

**Review by obrie — CHANGES_REQUESTED** [review]
One suggestion -- rest of it make sense

File: lib/newrelic-couchbase/instrumentation/couchbase.rb (hunk 1 above)
Comment (ydderd) [reply lib/newrelic-couchbase/instrumentation/couchbase.rb:4]: I think the `new` call was being recorded as `connect` below. Just moved it up here.

**Review by obrie — APPROVED** [review]

File: lib/newrelic-couchbase/instrumentation/bucket.rb (hunk 3 above)
Comment (obrie) [reply lib/newrelic-couchbase/instrumentation/bucket.rb:8]: Nevermind, misunderstood here

**Pull request merged**